src/shared/services/fields/fields.purger.ts

```diff
@@ -66,58 +66,227 @@ export class FieldsPurger {
     }
   }
 
-  private isHidden(condition, formFields) {
+  private isHidden(condition: ShowCondition, formFields: any): boolean {
     return !condition.match(formFields);
   }
 
-  private findCaseFieldByWizardPageFieldId(currentPage: WizardPage, wizardPageField: WizardPageField) {
+  private findCaseFieldByWizardPageFieldId(currentPage: WizardPage, wizardPageField: WizardPageField): CaseField {
     return currentPage.case_fields.find(cf => cf.id === wizardPageField.case_field_id);
   }
 
-  private hasShowConditionPage(wizardPage, formFields): boolean {
+  private hasShowConditionPage(wizardPage: WizardPage, formFields: any): boolean {
     return wizardPage.show_condition && formFields[this.getShowConditionKey(wizardPage.show_condition)];
   }
 
-  private hasShowConditionField(case_field, formFields): boolean {
+  private hasShowConditionField(case_field: CaseField, formFields: any): boolean {
     return case_field.show_condition && formFields.data[this.getShowConditionKey(case_field.show_condition)];
   }
 
-  private getShowConditionKey(show_condition): string {
-    return show_condition.split('=')[0];
+  private getShowConditionKey(show_condition: string): string {
+    // Need to allow for negated conditions, i.e. !=, as well as regular ones (=)
+    return show_condition.split(/!=|=/)[0];
```

Comment:
Should this also check for `'CONTAINS'`?

Reply:
@paul-graham - I have now added this